ec/ecresource/deploymentresource/elasticsearch/v1/schema.go

```diff
@@ -0,0 +1,544 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v1
+
+import (
+	"context"
+	"strings"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal/planmodifier"
+	"github.com/elastic/terraform-provider-ec/ec/internal/validators"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// These constants are only used to determine whether or not a dedicated
+// tier of masters or ingest (coordinating) nodes are set.
+const (
+	dataTierRolePrefix   = "data_"
+	ingestDataTierRole   = "ingest"
+	masterDataTierRole   = "master"
+	autodetect           = "autodetect"
+	growAndShrink        = "grow_and_shrink"
+	rollingGrowAndShrink = "rolling_grow_and_shrink"
+	rollingAll           = "rolling_all"
+)
+
+// List of update strategies availables.
+var strategiesList = []string{
+	autodetect, growAndShrink, rollingGrowAndShrink, rollingAll,
+}
+
+func ElasticsearchSchema() tfsdk.Attribute {
+	return tfsdk.Attribute{
+		Description: "Required Elasticsearch resource definition",
+		Required:    true,
+		Validators:  []tfsdk.AttributeValidator{listvalidator.SizeAtMost(1)},
+		Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
+			"autoscale": {
+				Type:        types.StringType,
+				Description: `Enable or disable autoscaling. Defaults to the setting coming from the deployment template. Accepted values are "true" or "false".`,
```

Comment:
Should this been a boolean attribute rather than a string?

Reply:
This is how it was before. But once we're changing the schema anyway, maybe it makes sense to convert it to boolean one.

Reply:
Done